lib/std/build.zig

```diff
@@ -2530,6 +2533,9 @@ pub const LibExeObjStep = struct {
             }
         } else switch (self.target.getExternalExecutor()) {
             .native, .unavailable => {},
+            .rosetta => if (self.enable_rosetta) {
+                try zig_args.append("--test-cmd-bin");
```

Comment:
Wouldn't this just be the same as `native`, i.e. no args appended?

Reply:
Yeah, I was wondering about that one, and I thought it was the case, but then note that `.native` and `.unavailable` share the prong which suggests they won't be run. In fact, I think in this case we could easily say here `.native => unreachable` since I guess it should have been handled in the first `if` block above?

Reply:
Hmm, no, scratch that, I think you're right. I think we can simply have `.native, .unavailable, .rosetta => {},`.

Reply:
OK, I just verified that:

```
$ zig test main.zig -target x86_64-macos
warning: created src/zig-cache/o/493cf4b056dc7e9721da37e880cfb846/test but skipping execution because it is non-native
```

while:

```
$ zig test main.zig -target x86_64-macos --test-cmd-bin
All 1 tests passed.
```

So I think we do want to check if Rosetta is enabled declaratively and then append `--test-cmd-bin` if so. Thoughts?